src/diffusers/models/unets/unet_stable_cascade.py

```diff
@@ -547,9 +547,11 @@ def custom_forward(*inputs):
                         if isinstance(block, SDCascadeResBlock):
                             skip = level_outputs[i] if k == 0 and i > 0 else None
                             if skip is not None and (x.size(-1) != skip.size(-1) or x.size(-2) != skip.size(-2)):
+                                orig_type = x.dtype
                                 x = torch.nn.functional.interpolate(
                                     x.float(), skip.shape[-2:], mode="bilinear", align_corners=True
                                 )
+                                x = x.type(orig_type)
```

Comment:
Could we use `x = x.to(orig_dtype` just to be consistent with how casting in done in other parts of the library.